### ISSUE TITLE
False positive for UnusedOperation with quote/unquote

### ISSUE THREAD

**Issue by michallepicki** [rrrene/credo#1270]
Hello! I am on credo `1.7.17`, elixir `1.19.4-otp-28`, erlang `28.3`

I am seeing unexpected warnings for some of our test helpers where a function is passed to a macro with `unquote`. I have reduced the example to:

```elixir
defmodule SomeMod do
  defmacro some_macro(some_fun) do
    quote do
      unquote(some_fun).(Map.merge(%{}, %{})

      :ok
    end
  end
end
```

Since the `Map.merge` result is used as an argument of the injected function, I think we shouldn't get a warning here

**Comment by rrrene** [comment]
@michallepicki Thx so much for reporting this! 🙏  Should now be fixed on `master` :+1:
